mlflow/deployments/base.py

```diff
@@ -182,22 +196,103 @@ def predict(self, deployment_name, df):
 
         :param deployment_name: Name of deployment to predict against
         :param df: Pandas DataFrame to use for inference
+        :param endpoint: Endpoint to predict against. May not be supported by all targets
         :return: A pandas DataFrame, pandas Series, or numpy array
         """
         pass
 
     @experimental
-    def explain(self, deployment_name, df):  # pylint: disable=unused-argument
+    def explain(
+        self, deployment_name=None, df=None, endpoint=None
+    ):  # pylint: disable=unused-argument
         """
         Generate explanations of model predictions on the specified input pandas Dataframe
         ``df`` for the deployed model. Explanation output formats vary by deployment target,
         and can include details like feature importance for understanding/debugging predictions.
 
         :param deployment_name: Name of deployment to predict against
         :param df: Pandas DataFrame to use for explaining feature importance in model prediction
+        :param endpoint: Endpoint to predict against. May not be supported by all targets
         :return: A JSON-able object (pandas dataframe, numpy array, dictionary), or
                  an exception if the implementation is not available in deployment target's class
         """
         raise MlflowException(
             "Computing model explanations is not yet supported for this deployment target"
         )
+
+    @abc.abstractmethod
```

Comment:
Decorating these with `@abc.abstractmethod` will break existing plugin implementations that inherit from `BaseDeploymentClient`, e.g.:

```
 Can't instantiate abstract class MyDeploymentClient with abstract methods create_endpoint, delete_endpoint, get_endpoint, list_endpoints, update_endpoint
```


I've confirmed that preexisting abstract method implementations that don't define the `endpoint` kwarg (e.g. the ones in SageMaker on `master`) continue to work just fine. However, I think we should keep the changes to SageMaker to fully comply with the new deployment client structure and incentivize contributors to add support for it.